Several tests fail by timeout

## TestTimeout

timeout without error or panic

## TestConnectivityOnStartup

timeout without error or panic

## TestAgentRestart/Restart_*

```
goroutine profile: total 18
8 @ 0x438af0 0x43181a 0x430d85 0x48d065 0x48de23 0x48de05 0x540c5b 0x558c3a 0x55778d 0x753f77 0x468031
#	0x430d84	internal/poll.runtime_pollWait+0x54			/usr/lib/golang/src/runtime/netpoll.go:203
#	0x48d064	internal/poll.(*pollDesc).wait+0x44			/usr/lib/golang/src/internal/poll/fd_poll_runtime.go:87
#	0x48de22	internal/poll.(*pollDesc).waitRead+0x1a2		/usr/lib/golang/src/internal/poll/fd_poll_runtime.go:92
#	0x48de04	internal/poll.(*FD).ReadFrom+0x184			/usr/lib/golang/src/internal/poll/fd_unix.go:219
#	0x540c5a	net.(*netFD).readFrom+0x5a				/usr/lib/golang/src/net/fd_unix.go:208
#	0x558c39	net.(*UDPConn).readFrom+0x69				/usr/lib/golang/src/net/udpsock_posix.go:47
#	0x55778c	net.(*UDPConn).ReadFrom+0x5c				/usr/lib/golang/src/net/udpsock.go:118
#	0x753f76	github.com/pion/ice/v2.(*candidateBase).recvLoop+0x116	/home/at-wat/go/src/github.com/pion/ice/candidate_base.go:92

2 @ 0x438af0 0x4077b7 0x4074bb 0x776d5c 0x468031
#	0x776d5b	github.com/pion/ice/v2.(*Agent).startOnConnectionStateChangeRoutine.func1+0x4b	/home/at-wat/go/src/github.com/pion/ice/agent.go:321

2 @ 0x438af0 0x43181a 0x430d85 0x48d065 0x48f066 0x48f043 0x551b78 0x58611e 0x58a1b7 0x58a18d 0x58e915 0x468031
#	0x430d84	internal/poll.runtime_pollWait+0x54			/usr/lib/golang/src/runtime/netpoll.go:203
#	0x48d064	internal/poll.(*pollDesc).wait+0x44			/usr/lib/golang/src/internal/poll/fd_poll_runtime.go:87
#	0x48f065	internal/poll.(*pollDesc).waitRead+0xf5			/usr/lib/golang/src/internal/poll/fd_poll_runtime.go:92
#	0x48f042	internal/poll.(*FD).RawRead+0xd2			/usr/lib/golang/src/internal/poll/fd_unix.go:534
#	0x551b77	net.(*rawConn).Read+0x67				/usr/lib/golang/src/net/rawconn.go:43
#	0x58611d	golang.org/x/net/internal/socket.(*Conn).recvMsg+0x20d	/home/at-wat/go/pkg/mod/golang.org/x/net@v0.0.0-20200602114024-627f9648deb9/internal/socket/rawconn_msg.go:32
#	0x58a1b6	golang.org/x/net/internal/socket.(*Conn).RecvMsg+0x1a6	/home/at-wat/go/pkg/mod/golang.org/x/net@v0.0.0-20200602114024-627f9648deb9/internal/socket/socket.go:255
#	0x58a18c	golang.org/x/net/ipv4.(*payloadHandler).ReadFrom+0x17c	/home/at-wat/go/pkg/mod/golang.org/x/net@v0.0.0-20200602114024-627f9648deb9/ipv4/payload_cmsg.go:31
#	0x58e914	github.com/pion/mdns.(*Conn).start+0x134		/home/at-wat/go/pkg/mod/github.com/pion/mdns@v0.0.4/conn.go:256

2 @ 0x438af0 0x448a53 0x74e8cd 0x468031
#	0x74e8cc	github.com/pion/ice/v2.(*Agent).connectivityChecks+0x12c	/home/at-wat/go/src/github.com/pion/ice/agent.go:402

1 @ 0x438af0 0x4077b7 0x40747b 0x4e716e 0x4eb0e8 0x4e6d7c 0x4e8677 0x4e75cf 0x7928b5 0x43872a 0x468031
#	0x4e716d	testing.(*T).Run+0x37d		/usr/lib/golang/src/testing/testing.go:1043
#	0x4eb0e7	testing.runTests.func1+0x77	/usr/lib/golang/src/testing/testing.go:1284
#	0x4e6d7b	testing.tRunner+0xdb		/usr/lib/golang/src/testing/testing.go:991
#	0x4e8676	testing.runTests+0x2a6		/usr/lib/golang/src/testing/testing.go:1282
#	0x4e75ce	testing.(*M).Run+0x15e		/usr/lib/golang/src/testing/testing.go:1199
#	0x7928b4	main.main+0x134			_testmain.go:154
#	0x438729	runtime.main+0x1f9		/usr/lib/golang/src/runtime/proc.go:203

1 @ 0x438af0 0x4077b7 0x40747b 0x4e716e 0x769e1a 0x4e6d7c 0x468031
#	0x4e716d	testing.(*T).Run+0x37d				/usr/lib/golang/src/testing/testing.go:1043
#	0x769e19	github.com/pion/ice/v2.TestAgentRestart+0x139	/home/at-wat/go/src/github.com/pion/ice/agent_test.go:1361
#	0x4e6d7b	testing.tRunner+0xdb				/usr/lib/golang/src/testing/testing.go:991

1 @ 0x438af0 0x4077b7 0x40747b 0x78154d 0x4e6d7c 0x468031
#	0x78154c	github.com/pion/ice/v2.TestAgentRestart.func4+0x46c	/home/at-wat/go/src/github.com/pion/ice/agent_test.go:1395
#	0x4e6d7b	testing.tRunner+0xdb					/usr/lib/golang/src/testing/testing.go:991

1 @ 0x520955 0x520770 0x51d50a 0x6dece9 0x468031
#	0x520954	runtime/pprof.writeRuntimeProfile+0x94			/usr/lib/golang/src/runtime/pprof/pprof.go:694
#	0x52076f	runtime/pprof.writeGoroutine+0x9f			/usr/lib/golang/src/runtime/pprof/pprof.go:656
#	0x51d509	runtime/pprof.(*Profile).WriteTo+0x3d9			/usr/lib/golang/src/runtime/pprof/pprof.go:329
#	0x6dece8	github.com/pion/transport/test.TimeOut.func1+0x68	/home/at-wat/go/pkg/mod/github.com/pion/transport@v0.10.0/test/util.go:18

panic: timeout
```

## TestHandlePeerReflexive

```
goroutine profile: total 15
6 @ 0x4386a0 0x43148a 0x430a55 0x48c915 0x48d6d3 0x48d6b5 0x54059b 0x55857a 0x5570cd 0x753887 0x4678e1
#	0x430a54	internal/poll.runtime_pollWait+0x54			/usr/lib/golang114/src/runtime/netpoll.go:203
#	0x48c914	internal/poll.(*pollDesc).wait+0x44			/usr/lib/golang114/src/internal/poll/fd_poll_runtime.go:87
#	0x48d6d2	internal/poll.(*pollDesc).waitRead+0x1a2		/usr/lib/golang114/src/internal/poll/fd_poll_runtime.go:92
#	0x48d6b4	internal/poll.(*FD).ReadFrom+0x184			/usr/lib/golang114/src/internal/poll/fd_unix.go:219
#	0x54059a	net.(*netFD).readFrom+0x5a				/usr/lib/golang114/src/net/fd_unix.go:208
#	0x558579	net.(*UDPConn).readFrom+0x69				/usr/lib/golang114/src/net/udpsock_posix.go:47
#	0x5570cc	net.(*UDPConn).ReadFrom+0x5c				/usr/lib/golang114/src/net/udpsock.go:118
#	0x753886	github.com/pion/ice/v2.(*candidateBase).recvLoop+0x116	/home/at-wat/go/src/github.com/pion/ice/candidate_base.go:92

2 @ 0x4386a0 0x407577 0x40727b 0x77666c 0x4678e1
#	0x77666b	github.com/pion/ice/v2.(*Agent).startOnConnectionStateChangeRoutine.func1+0x4b	/home/at-wat/go/src/github.com/pion/ice/agent.go:321

2 @ 0x4386a0 0x43148a 0x430a55 0x48c915 0x48e916 0x48e8f3 0x5514b8 0x585a2e 0x589ac7 0x589a9d 0x58e225 0x4678e1
#	0x430a54	internal/poll.runtime_pollWait+0x54			/usr/lib/golang114/src/runtime/netpoll.go:203
#	0x48c914	internal/poll.(*pollDesc).wait+0x44			/usr/lib/golang114/src/internal/poll/fd_poll_runtime.go:87
#	0x48e915	internal/poll.(*pollDesc).waitRead+0xf5			/usr/lib/golang114/src/internal/poll/fd_poll_runtime.go:92
#	0x48e8f2	internal/poll.(*FD).RawRead+0xd2			/usr/lib/golang114/src/internal/poll/fd_unix.go:534
#	0x5514b7	net.(*rawConn).Read+0x67				/usr/lib/golang114/src/net/rawconn.go:43
#	0x585a2d	golang.org/x/net/internal/socket.(*Conn).recvMsg+0x20d	/home/at-wat/go/pkg/mod/golang.org/x/net@v0.0.0-20200602114024-627f9648deb9/internal/socket/rawconn_msg.go:32
#	0x589ac6	golang.org/x/net/internal/socket.(*Conn).RecvMsg+0x1a6	/home/at-wat/go/pkg/mod/golang.org/x/net@v0.0.0-20200602114024-627f9648deb9/internal/socket/socket.go:255
#	0x589a9c	golang.org/x/net/ipv4.(*payloadHandler).ReadFrom+0x17c	/home/at-wat/go/pkg/mod/golang.org/x/net@v0.0.0-20200602114024-627f9648deb9/ipv4/payload_cmsg.go:31
#	0x58e224	github.com/pion/mdns.(*Conn).start+0x134		/home/at-wat/go/pkg/mod/github.com/pion/mdns@v0.0.4/conn.go:256

2 @ 0x4386a0 0x4484f3 0x74e1dd 0x4678e1
#	0x74e1dc	github.com/pion/ice/v2.(*Agent).connectivityChecks+0x12c	/home/at-wat/go/src/github.com/pion/ice/agent.go:402

1 @ 0x41be91 0x41be81 0x6de778 0x6dd860 0x7621e9 0x4e662c 0x4678e1
#	0x6de777	github.com/pion/transport/test.CheckRoutines.func1+0xd7	/home/at-wat/go/pkg/mod/github.com/pion/transport@v0.10.0/test/util.go:32
#	0x6dd85f	github.com/pion/transport/test.CheckRoutines+0x6f	/home/at-wat/go/pkg/mod/github.com/pion/transport@v0.10.0/test/util.go:44
#	0x7621e8	github.com/pion/ice/v2.TestHandlePeerReflexive+0x68	/home/at-wat/go/src/github.com/pion/ice/agent_test.go:226
#	0x4e662b	testing.tRunner+0xdb					/usr/lib/golang114/src/testing/testing.go:992

1 @ 0x4386a0 0x407577 0x40723b 0x4e6a1e 0x4eaa78 0x4e662c 0x4e7f27 0x4e6e7f 0x7921c5 0x4382c2 0x4678e1
#	0x4e6a1d	testing.(*T).Run+0x37d		/usr/lib/golang114/src/testing/testing.go:1044
#	0x4eaa77	testing.runTests.func1+0x77	/usr/lib/golang114/src/testing/testing.go:1285
#	0x4e662b	testing.tRunner+0xdb		/usr/lib/golang114/src/testing/testing.go:992
#	0x4e7f26	testing.runTests+0x2a6		/usr/lib/golang114/src/testing/testing.go:1283
#	0x4e6e7e	testing.(*M).Run+0x15e		/usr/lib/golang114/src/testing/testing.go:1200
#	0x7921c4	main.main+0x134			_testmain.go:154
#	0x4382c1	runtime.main+0x211		/usr/lib/golang114/src/runtime/proc.go:203

1 @ 0x5202b5 0x5200d0 0x51ce9a 0x6de5f9 0x4678e1
#	0x5202b4	runtime/pprof.writeRuntimeProfile+0x94			/usr/lib/golang114/src/runtime/pprof/pprof.go:694
#	0x5200cf	runtime/pprof.writeGoroutine+0x9f			/usr/lib/golang114/src/runtime/pprof/pprof.go:656
#	0x51ce99	runtime/pprof.(*Profile).WriteTo+0x3d9			/usr/lib/golang114/src/runtime/pprof/pprof.go:329
#	0x6de5f8	github.com/pion/transport/test.TimeOut.func1+0x68	/home/at-wat/go/pkg/mod/github.com/pion/transport@v0.10.0/test/util.go:18

panic: timeout
```

## TestConnectionStateFailedDeleteAllCandidates

```
goroutine profile: total 9
2 @ 0x438af0 0x4077b7 0x4074bb 0x776d5c 0x468031
#	0x776d5b	github.com/pion/ice/v2.(*Agent).startOnConnectionStateChangeRoutine.func1+0x4b	/home/at-wat/go/src/github.com/pion/ice/agent.go:321

2 @ 0x438af0 0x43181a 0x430d85 0x48d065 0x48f066 0x48f043 0x551b78 0x58611e 0x58a1b7 0x58a18d 0x58e915 0x468031
#	0x430d84	internal/poll.runtime_pollWait+0x54			/usr/lib/golang/src/runtime/netpoll.go:203
#	0x48d064	internal/poll.(*pollDesc).wait+0x44			/usr/lib/golang/src/internal/poll/fd_poll_runtime.go:87
#	0x48f065	internal/poll.(*pollDesc).waitRead+0xf5			/usr/lib/golang/src/internal/poll/fd_poll_runtime.go:92
#	0x48f042	internal/poll.(*FD).RawRead+0xd2			/usr/lib/golang/src/internal/poll/fd_unix.go:534
#	0x551b77	net.(*rawConn).Read+0x67				/usr/lib/golang/src/net/rawconn.go:43
#	0x58611d	golang.org/x/net/internal/socket.(*Conn).recvMsg+0x20d	/home/at-wat/go/pkg/mod/golang.org/x/net@v0.0.0-20200602114024-627f9648deb9/internal/socket/rawconn_msg.go:32
#	0x58a1b6	golang.org/x/net/internal/socket.(*Conn).RecvMsg+0x1a6	/home/at-wat/go/pkg/mod/golang.org/x/net@v0.0.0-20200602114024-627f9648deb9/internal/socket/socket.go:255
#	0x58a18c	golang.org/x/net/ipv4.(*payloadHandler).ReadFrom+0x17c	/home/at-wat/go/pkg/mod/golang.org/x/net@v0.0.0-20200602114024-627f9648deb9/ipv4/payload_cmsg.go:31
#	0x58e914	github.com/pion/mdns.(*Conn).start+0x134		/home/at-wat/go/pkg/mod/github.com/pion/mdns@v0.0.4/conn.go:256

2 @ 0x438af0 0x448a53 0x74e8cd 0x468031
#	0x74e8cc	github.com/pion/ice/v2.(*Agent).connectivityChecks+0x12c	/home/at-wat/go/src/github.com/pion/ice/agent.go:402

1 @ 0x438af0 0x4077b7 0x40747b 0x4e716e 0x4eb0e8 0x4e6d7c 0x4e8677 0x4e75cf 0x7928b5 0x43872a 0x468031
#	0x4e716d	testing.(*T).Run+0x37d		/usr/lib/golang/src/testing/testing.go:1043
#	0x4eb0e7	testing.runTests.func1+0x77	/usr/lib/golang/src/testing/testing.go:1284
#	0x4e6d7b	testing.tRunner+0xdb		/usr/lib/golang/src/testing/testing.go:991
#	0x4e8676	testing.runTests+0x2a6		/usr/lib/golang/src/testing/testing.go:1282
#	0x4e75ce	testing.(*M).Run+0x15e		/usr/lib/golang/src/testing/testing.go:1199
#	0x7928b4	main.main+0x134			_testmain.go:154
#	0x438729	runtime.main+0x1f9		/usr/lib/golang/src/runtime/proc.go:203

1 @ 0x438af0 0x448a53 0x75da00 0x775746 0x7756df 0x7695f8 0x4e6d7c 0x468031
#	0x75d9ff	github.com/pion/ice/v2.(*Agent).connect+0x18f				/home/at-wat/go/src/github.com/pion/ice/transport.go:57
#	0x775745	github.com/pion/ice/v2.(*Agent).Dial+0x125				/home/at-wat/go/src/github.com/pion/ice/transport.go:16
#	0x7756de	github.com/pion/ice/v2.connect+0xbe					/home/at-wat/go/src/github.com/pion/ice/transport_test.go:209
#	0x7695f7	github.com/pion/ice/v2.TestConnectionStateFailedDeleteAllCandidates+0x2c7	/home/at-wat/go/src/github.com/pion/ice/agent_test.go:1239
#	0x4e6d7b	testing.tRunner+0xdb							/usr/lib/golang/src/testing/testing.go:991

1 @ 0x520955 0x520770 0x51d50a 0x6dece9 0x468031
#	0x520954	runtime/pprof.writeRuntimeProfile+0x94			/usr/lib/golang/src/runtime/pprof/pprof.go:694
#	0x52076f	runtime/pprof.writeGoroutine+0x9f			/usr/lib/golang/src/runtime/pprof/pprof.go:656
#	0x51d509	runtime/pprof.(*Profile).WriteTo+0x3d9			/usr/lib/golang/src/runtime/pprof/pprof.go:329
#	0x6dece8	github.com/pion/transport/test.TimeOut.func1+0x68	/home/at-wat/go/pkg/mod/github.com/pion/transport@v0.10.0/test/util.go:18

panic: timeout
```

TestAgentRestart still causes timeout.
```
=== RUN   TestAgentRestart/Restart_One_Side
goroutine profile: total 11
2 @ 0x4655a0 0x433818 0x4335cb 0x99b419 0x4964e1
#	0x99b418	github.com/pion/ice/v2.(*Agent).startOnConnectionStateChangeRoutine.func1+0x88	/home/travis/gopath/src/github.com/pion/ice/agent.go:322

2 @ 0x4655a0 0x45e36a 0x45d935 0x4d6934 0x4d993e 0x4d9907 0x637d6f 0x688512 0x68f3f1 0x68f3b2 0x696485 0x4964e1
#	0x45d934	internal/poll.runtime_pollWait+0x54			/home/travis/.gimme/versions/go1.14.linux.amd64/src/runtime/netpoll.go:203
#	0x4d6933	internal/poll.(*pollDesc).wait+0xe3			/home/travis/.gimme/versions/go1.14.linux.amd64/src/internal/poll/fd_poll_runtime.go:87
#	0x4d993d	internal/poll.(*pollDesc).waitRead+0x17d		/home/travis/.gimme/versions/go1.14.linux.amd64/src/internal/poll/fd_poll_runtime.go:92
#	0x4d9906	internal/poll.(*FD).RawRead+0x146			/home/travis/.gimme/versions/go1.14.linux.amd64/src/internal/poll/fd_unix.go:534
#	0x637d6e	net.(*rawConn).Read+0x6e				/home/travis/.gimme/versions/go1.14.linux.amd64/src/net/rawconn.go:43
#	0x688511	golang.org/x/net/internal/socket.(*Conn).recvMsg+0x3e1	/home/travis/gopath/pkg/mod/golang.org/x/net@v0.0.0-20200602114024-627f9648deb9/internal/socket/rawconn_msg.go:32
#	0x68f3f0	golang.org/x/net/internal/socket.(*Conn).RecvMsg+0x210	/home/travis/gopath/pkg/mod/golang.org/x/net@v0.0.0-20200602114024-627f9648deb9/internal/socket/socket.go:255
#	0x68f3b1	golang.org/x/net/ipv4.(*payloadHandler).ReadFrom+0x1d1	/home/travis/gopath/pkg/mod/golang.org/x/net@v0.0.0-20200602114024-627f9648deb9/ipv4/payload_cmsg.go:31
#	0x696484	github.com/pion/mdns.(*Conn).start+0x174		/home/travis/gopath/pkg/mod/github.com/pion/mdns@v0.0.4/conn.go:256

2 @ 0x4655a0 0x4768e9 0x95b45e 0x4964e1
#	0x95b45d	github.com/pion/ice/v2.(*Agent).connectivityChecks+0x24d	/home/travis/gopath/src/github.com/pion/ice/agent.go:407

1 @ 0x4655a0 0x433818 0x43358b 0x57fe09 0x5862a7 0x57f6bc 0x581d78 0x5805d0 0x9c6508 0x4651a2 0x4964e1
#	0x57fe08	testing.(*T).Run+0x698		/home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:1044
#	0x5862a6	testing.runTests.func1+0xa6	/home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:1285
#	0x57f6bb	testing.tRunner+0x1eb		/home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:992
#	0x581d77	testing.runTests+0x527		/home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:1283
#	0x5805cf	testing.(*M).Run+0x2ff		/home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:1200
#	0x9c6507	main.main+0x347			_testmain.go:266
#	0x4651a1	runtime.main+0x211		/home/travis/.gimme/versions/go1.14.linux.amd64/src/runtime/proc.go:203

1 @ 0x4655a0 0x433818 0x43358b 0x57fe09 0x988cdc 0x57f6bc 0x4964e1
#	0x57fe08	testing.(*T).Run+0x698				/home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:1044
#	0x988cdb	github.com/pion/ice/v2.TestAgentRestart+0x15b	/home/travis/gopath/src/github.com/pion/ice/agent_test.go:1388
#	0x57f6bb	testing.tRunner+0x1eb				/home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:992

1 @ 0x4655a0 0x4768e9 0x978469 0x977f6b 0x999594 0x999933 0x9aa6da 0x57f6bc 0x4964e1
#	0x978468	github.com/pion/ice/v2.(*Agent).connect+0x278		/home/travis/gopath/src/github.com/pion/ice/transport.go:54
#	0x977f6a	github.com/pion/ice/v2.(*Agent).Dial+0xaa		/home/travis/gopath/src/github.com/pion/ice/transport.go:16
#	0x999593	github.com/pion/ice/v2.connect+0x1f3			/home/travis/gopath/src/github.com/pion/ice/transport_test.go:236
#	0x999932	github.com/pion/ice/v2.pipe+0x2b2			/home/travis/gopath/src/github.com/pion/ice/transport_test.go:267
#	0x9aa6d9	github.com/pion/ice/v2.TestAgentRestart.func3+0xd9	/home/travis/gopath/src/github.com/pion/ice/agent_test.go:1390
#	0x57f6bb	testing.tRunner+0x1eb					/home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:992

1 @ 0x4655a0 0x4768e9 0x978469 0x97807b 0x9ba654 0x4964e1
#	0x978468	github.com/pion/ice/v2.(*Agent).connect+0x278	/home/travis/gopath/src/github.com/pion/ice/transport.go:54
#	0x97807a	github.com/pion/ice/v2.(*Agent).Accept+0xaa	/home/travis/gopath/src/github.com/pion/ice/transport.go:22
#	0x9ba653	github.com/pion/ice/v2.connect.func1+0x123	/home/travis/gopath/src/github.com/pion/ice/transport_test.go:230

1 @ 0x5ed523 0x5ed2ea 0x5e942c 0x8aa4b4 0x4964e1
#	0x5ed522	runtime/pprof.writeRuntimeProfile+0xa2			/home/travis/.gimme/versions/go1.14.linux.amd64/src/runtime/pprof/pprof.go:694
#	0x5ed2e9	runtime/pprof.writeGoroutine+0xc9			/home/travis/.gimme/versions/go1.14.linux.amd64/src/runtime/pprof/pprof.go:656
#	0x5e942b	runtime/pprof.(*Profile).WriteTo+0x4fb			/home/travis/.gimme/versions/go1.14.linux.amd64/src/runtime/pprof/pprof.go:329
#	0x8aa4b3	github.com/pion/transport/test.TimeOut.func1+0x93	/home/travis/gopath/pkg/mod/github.com/pion/transport@v0.10.0/test/util.go:18

panic: timeout
```

--- a/agent_test.go
+++ b/agent_test.go
@@ -1367,6 +1367,8 @@ func TestAgentRestart(t *testing.T) {
 	lim := test.TimeOut(time.Second * 30)
 	defer lim.Stop()
 
+	oneSecond := time.Second
+
 	t.Run("Restart During Gather", func(t *testing.T) {
 		agent, err := NewAgent(&AgentConfig{})
 		assert.NoError(t, err)
@@ -1386,10 +1388,10 @@ func TestAgentRestart(t *testing.T) {
 	})
 
 	t.Run("Restart One Side", func(t *testing.T) {
-		oneSecond := time.Second
 		connA, connB := pipe(&AgentConfig{
 			DisconnectedTimeout: &oneSecond,
 			FailedTimeout:       &oneSecond,
+			taskLoopInterval:    50 * time.Millisecond,
 		})
 
 		ctx, cancel := context.WithCancel(context.Background())
@@ -1418,7 +1420,11 @@ func TestAgentRestart(t *testing.T) {
 		}
 
 		// Store the original candidates, confirm that after we reconnect we have new pairs
-		connA, connB := pipe(nil)
+		connA, connB := pipe(&AgentConfig{
+			DisconnectedTimeout: &oneSecond,
+			FailedTimeout:       &oneSecond,
+			taskLoopInterval:    50 * time.Millisecond,
+		})
 		connAFirstCandidates := generateCandidateAddressStrings(connA.agent.GetLocalCandidates())
 		connBFirstCandidates := generateCandidateAddressStrings(connB.agent.GetLocalCandidates())
 

